src/details/ArborX_MinimumSpanningTree.hpp

```diff
@@ -174,20 +174,26 @@ struct FindComponentNearestNeighbors
   {
     constexpr auto inf = KokkosExt::ArithmeticTraits::infinity<float>::value;
 
-    auto const distance = [bounding_volume_i =
-                               HappyTreeFriends::getBoundingVolume(_bvh, i),
-                           &bvh = _bvh](int j) {
-      using Details::distance;
-      return distance(bounding_volume_i,
-                      HappyTreeFriends::getBoundingVolume(bvh, j));
-    };
+    auto const distance =
+        [bounding_volume_i = HappyTreeFriends::getBoundingVolume(
+             LeafNodeTag{}, _bvh, HappyTreeFriends::leafIndex(_bvh, i)),
+         &bvh = _bvh](int j) {
```

Comment:
readability
```C++
auto const bounding_volume_j = ...

Reply:
I don't understand.

Reply:
Wouldn't that prevent some of the potential optimizations depending on what `getBoundingVolume` returns and how the `distance` computes? Like, by value, by reference, etc?